README.md

```diff
@@ -1,2 +1,158 @@
-# Super dooper cool voice + text configurator
-More to come soon!
+# Contoso Voice Agent
+
+A sophisticated AI-powered voice and text assistant for Contoso Outdoors that provides personalized product recommendations and customer support through natural conversations.
+
+## Project Overview
+
+The Contoso Voice Agent combines text chat and voice calling capabilities to help customers discover and purchase outdoor gear. It serves as a retail assistant for Contoso Outdoor Company, creating an interactive and responsive shopping experience through AI-driven conversations.
+
+## Architecture
+
+```mermaid
+graph TD
+    subgraph Frontend["Frontend (Next.js)"]
+        UI["UI Components"]
+        Chat["Chat Interface"]
+        Voice["Voice Interface"]
+        WebAudio["Web Audio API"]
+        Store["State Management (Zustand)"]
+        WebSocketClient["WebSocket Client"]
+    end
+
+    subgraph Backend["Backend (FastAPI)"]
+        API["API Endpoints"]
+        Session["Session Management"]
+        WSChat["WebSocket Chat Handler"]
+        WSVoice["WebSocket Voice Handler"]
+        Suggestions["Product Suggestions"]
+    end
+
+    subgraph Azure["Azure OpenAI"]
+        GPT4o["GPT-4o"]
+        Realtime["Realtime Voice API"]
+    end
+
+    subgraph Data["Data"]
+        Products["Product Catalog"]
+        Purchases["Purchase History"]
+    end
+
+    WebSocketClient -- "WebSocket (Text)" --> WSChat
+    WebSocketClient -- "WebSocket (Audio)" --> WSVoice
+    WebAudio -- "Audio Stream" --> WebSocketClient
+    
+    WSChat --> Session
+    WSVoice --> Session
+    
+    Session --> GPT4o
+    WSVoice --> Realtime
+    
+    Products --> Suggestions
+    Purchases --> Suggestions
+    Suggestions --> GPT4o
+    
+    API -- "HTTP" --> Frontend
+    
+    Store -- "State" --> UI
+    Store -- "State" --> Chat
+    Store -- "State" --> Voice
+```
+
+## Components
+
+### Frontend (web/)
+
+- **Chat Interface**: Real-time text conversations with AI assistant
+- **Voice Interface**: Interactive voice calling system with real-time audio processing
+- **Web Audio API Integration**: Audio capture, processing, and playback
+- **State Management**: Zustand stores for global state
+- **WebSocket Communication**: Real-time message exchange with backend
+- **UI Components**: React components with TypeScript for the product catalog and interactions
+
+### Backend (api/)
+
+- **FastAPI Application**: Main backend service with WebSocket support
+- **Session Management**: Maintains conversation context and user state
```

Comment:
I think it makes more sense to say it maintains the socket to the web client and the realtime API client - I don't there's any other state here